### PR TITLE
add primaryUAI support on workspace

### PR DESCRIPTION
**Pull request by zhaomuzhi**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

*
*
*

### CODE REVIEW

**Comment by guanyu-240** [comment]
@bmoore-msft Please review. Thank you!

File: 201-machine-learning-advanced/azuredeploy.json [hunk 1]
```diff
@@ -521,6 +540,47 @@
         "hbiWorkspace": "[parameters('confidential_data')]"
       }
     },
+    {
```
Comment (bmoore-msft) [inline 201-machine-learning-advanced/azuredeploy.json:543]: A couple things here:
1) remove the duplicate resource definition
2) using conditions only on the properties that differ

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

File: 201-machine-learning-advanced/azuredeploy.json (hunk 1 above)
Comment (zhaomuzhi) [reply 201-machine-learning-advanced/azuredeploy.json:543]: Hi, Brian, from my understanding on this doc, https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/conditional-resource-deployment, condition is only supported on resource level. Would you please show me an example about how to conditionally add new properties into a resource?

Comment (zhaomuzhi) [reply 201-machine-learning-advanced/azuredeploy.json]: addressed

**Comment by bmoore-msft** [comment]
@Blackmist - check this PR for updates that may be needed to documentation that references this sample.

**Comment by bmoore-msft** [comment]
@Blackmist - check this PR for updates that may be needed to documentation that references this sample.

**Comment by bmoore-msft** [comment]
@Blackmist - check this PR for updates that may be needed to documentation that references this sample.

**Comment by bmoore-msft** [comment]
@Blackmist - check this PR for updates that may be needed to documentation that references this sample.

**Comment by bmoore-msft** [comment]
@Blackmist - check this PR for updates that may be needed to documentation that references this sample.

**Comment by bmoore-msft** [comment]
@Blackmist - check this PR for updates that may be needed to documentation that references this sample.

File: 201-machine-learning-advanced/azuredeploy.json [hunk 2]
```diff
@@ -134,6 +166,13 @@
         "description": "Name of ApplicationInsights."
       }
     },
+    "appInsightLocation": {
+      "type": "string",
+      "defaultValue": "[if(or(equals(parameters('location'),'westcentralus'), contains(parameters('location'), 'euap')),'southcentralus',parameters('location'))]",
```
Comment (bmoore-msft) [inline 201-machine-learning-advanced/azuredeploy.json:171]: remove the if() hear and use a location object var to map the desired location - you can remove euap from the map.

File: 201-machine-learning-advanced/azuredeploy.json [hunk 3]
```diff
@@ -25,6 +25,24 @@
...
+      ],
+      "metadata": {
+        "description": "Specifies the identity type of the Azure Machine Learning workspace."
+      }
+    },
+    "primaryUserAssignedIdentity": {
+      "type": "string",
+      "defaultValue": "",
+      "metadata": {
+        "description": "Specifies the user assigned identity that represents the Azure Machine Learing workspace."
```
Comment (bmoore-msft) [inline 201-machine-learning-advanced/azuredeploy.json:43]: is this the name or the resourceId?  If the latter use separate parameters for sub,rg,name

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

File: 201-machine-learning-advanced/azuredeploy.json (hunk 2 above)
Comment (zhaomuzhi) [reply 201-machine-learning-advanced/azuredeploy.json:171]: Sorry, I didn't get it, would you please elaborate? Do you want me to create an variable and reference it here? I don't think I can reference variable in parameters section...

File: 201-machine-learning-advanced/azuredeploy.json (hunk 3 above)
Comment (zhaomuzhi) [reply 201-machine-learning-advanced/azuredeploy.json]: separate the parameters for rg and name

**Comment by bmoore-msft** [comment]
@Blackmist - check this PR for updates that may be needed to documentation that references this sample.  [this is an automated message]

File: 201-machine-learning-advanced/azuredeploy.json (hunk 2 above)
Comment (bmoore-msft) [reply 201-machine-learning-advanced/azuredeploy.json:178]: See: https://github.com/Azure/azure-quickstart-templates/blob/master/azmgmt-demo/azuredeploy.json#L72

File: 201-machine-learning-advanced/azuredeploy.json [hunk 4]
```diff
@@ -50,6 +75,13 @@
         "description": "Name of the storage account."
       }
     },
+    "storageLocation": {
```
Comment (bmoore-msft) [inline 201-machine-learning-advanced/azuredeploy.json:78]: remove this param - the common pattern (by far) is that these will be co-located.

File: 201-machine-learning-advanced/azuredeploy.json [hunk 5]
```diff
@@ -98,6 +130,13 @@
         "description": "Name of the key vault."
       }
     },
+    "keyVaultLocation": {
```
Comment (bmoore-msft) [inline 201-machine-learning-advanced/azuredeploy.json:133]: remove this param

File: 201-machine-learning-advanced/azuredeploy.json [hunk 6]
```diff
@@ -160,6 +206,13 @@
         "description": "The container registry bind to the workspace."
       }
     },
+    "containerRegistryLocation": {
```
Comment (bmoore-msft) [inline 201-machine-learning-advanced/azuredeploy.json:209]: remove this param

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

**Comment by bmoore-msft** [comment]
@Blackmist - check this PR for updates that may be needed to documentation that references this sample.  [this is an automated message]

**Pull request merged**